Master

Parthpisat opened the pull request:
Connected the flow & flow_record with logger to record the steps of flow.
Also done some changes in the overall code to connect the new files.

Parthpisat committed:
Add existing project files

Parthpisat committed:
The code is running and generating logs

Parthpisat committed:
Done all the changes as requested

rkrishnasanka committed:
Made the basic changes to address all the code quality issues and removed the dependencies

Co-authored-by: Parth <Parthpisat@users.noreply.github.com>

rkrishnasanka committed:
Added a datamodel for the the mongodb connection

Parthpisat committed:
Configured the database using the datamodel with the help of pydantic lib

Parthpisat committed:
Configured flow_record model to record the flow of data

rkrishnasanka commented on ml_workflow_logger/logger.py:
```diff
@@ -0,0 +1,197 @@
+            step_info (str): _description_. Defaults to None
+        """
+        step = Step(step_name=step_name, step_info=step_info)
+        flow = Flow(name="current_flow")
```
Update this to take the flow name from the `start_run()` method

rkrishnasanka commented on ml_workflow_logger/logger.py:
```diff
@@ -0,0 +1,197 @@
+            cls._instance.__init__(*args, **kwargs)
+        return cls._instance
+
+    def __init__(self, log_dir: Path=Path('logs'), graph: Optional[nx.DiGraph]=None, db_config: Optional[DBConfig]=None):
```
Don't take the graph here because we only have a single logger instance globally

rkrishnasanka replied:
Create a method to add flows

Parthpisat committed:
Logger is complete and ready to use

rkrishnasanka commented on ml_workflow_logger/models/flow.py:
```diff
@@ -0,0 +1,30 @@
+# Class representing the flow of the ML workflow
+class Flow(BaseModel):
+    id: str = Field(default_factory=lambda: str(uuid.uuid4()), alias='_id')
+    name: str = ""
```
Are things saving correctly even without the field ?

rkrishnasanka commented on ml_workflow_logger/models/flow_record.py:
```diff
@@ -0,0 +1,18 @@
+
+class FlowRecord(BaseModel):
+   id: str = Field(default_factory=lambda: str(uuid.uuid4()), alias='_id')
+    # TODO: Fill out the other records you need to have to make this a standardized record
```
Are these still relevant ?

rkrishnasanka commented on ml_workflow_logger/logger.py:
```diff
@@ -0,0 +1,279 @@
+                raise ValueError("No active run. Start a run before adding flows")
+            
+            # Create the new flow
+            new_flow = Flow(name=flow_name, data=flow_data or [])
```
There a problem here, there is no argument called data anymore

rkrishnasanka commented on ml_workflow_logger/run.py:
```diff
@@ -0,0 +1,35 @@
+from datetime import datetime
+
+class Run:
+    def __init__(self, run_name: str):
```
@Parthpisat where are doc strings ?

rkrishnasanka commented on ml_workflow_logger/logger.py:
```diff
@@ -0,0 +1,279 @@
+import sys
+sys.path.append('/home/venom983/Desktop/ml-workflow-logger')
+from typing import Any, Dict, Optional
```
Why do you have this ? If you're using there's something wrong with the way you're testing stuff

rkrishnasanka committed:
Started making the baseline changes. Primarily:

1. We need to improve the tests
2. We need to seperate logic classes vs data model classes
3. Its ambigous how each of the models interact with each other (missing data model diagram)

rkrishnasanka commented on ml_workflow_logger/logger.py:
```diff
@@ -0,0 +1,288 @@
+            self.local_store.save_step(self._current_run.run_id, flow_record.to_dict())
+
+            # Log to global database if available
+            if self.global_collection:
```
I think it might make better sense to have a global mongodb connection and then have helper methods to save all the data to mongodb

rkrishnasanka commented on ml_workflow_logger/logger.py:
```diff
@@ -0,0 +1,288 @@
+            flow_record =FlowRecord(
+                step_name="init",
+                step_data={"info": f"Flow {flow_name} initialized"},
+                flow_ref=new_flow,
```
This needs to be pulled from the mongodb collection. Its better to not have this logic here

rkrishnasanka committed:
Refactored the codebase halfway to have abstract drivers, etc.

Main issue stands that the division of logic is not clean and there needs to be type enforced message passing and only the data that gets saved onto mongodb or one of the other dabases needs to use the models

rkrishnasanka committed:
Added a README with system design

Co-authored-by: Parth <Parthpisat@users.noreply.github.com>

Parthpisat committed:
Separated logic from main file and made separate folder files for all models and interconnected them.

rkrishnasanka commented on ml_workflow_logger/logger.py:
```diff
@@ -0,0 +1,126 @@
+    def log_run(self, run_name: str, run_id: Optional[str]) -> None:
+        """Log run object, pass to driver for model conversion."""
+        try:
+            run_data = Run(run_name, run_id)
```
This needs to get the object and assign it here

rkrishnasanka committed:
Deleted duplicates and generated files from source tree

rkrishnasanka reviewed (changes requested):
This project needs comprehensive Unit tests using pytest. Please fix this ASAP. This project is already 4 weeks delayed at this point.

Also update the models readme with the actual descriptions of the datamodels, etc. The outside readme should have examples of how this library should be used, etc.

Parthpisat committed:
Need to resolve a return parameter issue in test and experiment file

Parthpisat committed:
Merge remote-tracking branch 'origin/master'

rkrishnasanka committed:
Fixed issue where the __new__ method was not accepting the arguments for the singleton class.

Passing of arguments should most likely replace the existing instance, so perhaps attaching the db driver and stuff differently might make sense.

Parthpisat committed:
Updating the latest version

rkrishnasanka committed:
Removed junk csv files

rkrishnasanka committed:
Ran black and isort

rkrishnasanka committed:
Refactored the local API and took care of all immediate errors

rkrishnasanka committed:
Updated the local run example

rkrishnasanka committed:
Made some more changes in the logger file to differentiate between local and cloud behaviors.

Parthpisat committed:
Co-authored-by: nandakishorekakarla21@outlook.com <nandakishorekakarla21@outlook.com>
Co-authored-by: Radhakrishna Sanka <rkrishnasanka@gmail.com>

Parthpisat committed:
Made some more changes in logger file and made sure local part of code is generating accurate output
